crates/provider/src/provider/trait.rs

```diff
@@ -650,24 +650,14 @@ pub trait Provider<T: Transport + Clone = BoxTransport, N: Network = Ethereum>:
         self.client().request("eth_syncing", ()).await
     }
 
-    /// Execute a smart contract call with a transaction request, without publishing a transaction.
-    async fn call(&self, tx: &N::TransactionRequest, block: BlockId) -> TransportResult<Bytes> {
-        self.client().request("eth_call", (tx, block)).await
-    }
-
     /// Execute a smart contract call with a transaction request and state overrides, without
     /// publishing a transaction.
     ///
     /// # Note
     ///
     /// Not all client implementations support state overrides.
-    async fn call_with_overrides(
-        &self,
-        tx: &N::TransactionRequest,
-        block: BlockId,
-        state: StateOverride,
-    ) -> TransportResult<Bytes> {
-        self.client().request("eth_call", (tx, block, state)).await
+    fn call<'req>(&self, tx: &'req N::TransactionRequest) -> EthCall<'req, 'static, T, N> {
```

Comment:
maybe add an example and also an alias for `call_with_overrides` so it comes up on docs.rs?